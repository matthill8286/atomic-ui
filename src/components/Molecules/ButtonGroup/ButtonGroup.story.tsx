/* eslint-disable @typescript-eslint/ban-ts-comment */
// @ts-nocheck => JSX property style does not exists in Typo component
import * as React from 'react'
import { withState } from '@dump247/storybook-state'
import { storiesOf } from '@storybook/react'
import { ButtonGroup } from './ButtonGroup'
import { BorderedButton } from './ButtonGroup.styled'
import { Button } from '@/components/Atoms/Button'
import { Typo } from '@/components/Atoms/Typography/Typo'
import { styled } from '@/styles/styled'

const stories = storiesOf('Design System/Molecules/ButtonGroup', module)

const voidFn = () => {}
const StyledDivWithTopBorder = styled.div`
  padding-top: 25px;
  position: relative;
  background-color: #fff;
  border-top: 1px solid #000;

  &:after,
  &:before {
    top: -1px;
    left: 7px;
    border: solid transparent;
    content: '';
    height: 0;
    width: 0;
    position: absolute;
    pointer-events: none;
  }

  &:after {
    border-color: rgba(136, 183, 213, 0);
    border-top-color: #fff;
    border-width: 8px;
    margin-left: -8px;
  }
  &:before {
    border-color: rgba(194, 225, 245, 0);
    border-top-color: #000;
    border-width: 9px;
    margin-left: -9px;
  }
`

stories.add(
  'With 2 Buttons',
  withState({ value: '1' }, (store) => {
    return (
      <div style={{ padding: '20px' }}>
        <ButtonGroup>
          <Button actionType="outlined" onClick={voidFn} size="sm" type="button">
            Button 1
          </Button>
          <Button actionType="outlined" onClick={voidFn} size="sm" type="button">
            Button 2
          </Button>
        </ButtonGroup>
      </div>
    )
  })
)

stories.add(
  'With 3 Buttons',
  withState({ value: '1' }, (store) => {
    return (
      <div style={{ padding: '20px' }}>
        <ButtonGroup>
          <Button actionType="outlined" onClick={voidFn} size="sm" type="button">
            Button 1
          </Button>
          <Button actionType="outlined" onClick={voidFn} size="sm" type="button">
            Button 2
          </Button>
          <Button actionType="outlined" onClick={voidFn} size="sm" type="button">
            Button 3
          </Button>
        </ButtonGroup>
        <br />
        <ButtonGroup>
          <Button actionType="secondary" onClick={voidFn} size="sm" type="button">
            Button 1
          </Button>
          <Button actionType="secondary" onClick={voidFn} size="sm" type="button">
            Button 2
          </Button>
          <Button actionType="secondary" onClick={voidFn} size="sm" type="button">
            Button 3
          </Button>
        </ButtonGroup>
        <br />
        <ButtonGroup>
          <Button actionType="prominent" onClick={voidFn} size="sm" type="button">
            Button 1
          </Button>
          <Button actionType="prominent" onClick={voidFn} size="sm" type="button">
            Button 2
          </Button>
          <Button actionType="prominent" onClick={voidFn} size="sm" type="button">
            Button 3
          </Button>
        </ButtonGroup>
      </div>
    )
  })
)

stories.add(
  'Example #1 - Delivery slots',
  withState({ value: '0' }, (store) => {
    return (
      <div style={{ padding: '20px' }}>
        <StyledDivWithTopBorder>
          <Typo tag="div" fontSize="xs" weight="bold" style={{ marginBottom: '7px' }}>
            Escoge el día y hora que desees:
          </Typo>
          <Typo tag="div" fontSize="xs" weight="regular" style={{ marginBottom: '4px' }}>
            Jueves 24.09
          </Typo>
          <ButtonGroup>
            <BorderedButton
              type="button"
              selected={store.state.value === '1'}
              onClick={() => store.set({ value: '1' })}>
              <Typo tag="span" fontSize="xxs" weight="regular">
                07—12h
              </Typo>
            </BorderedButton>
            <BorderedButton
              type="button"
              selected={store.state.value === '2'}
              onClick={() => store.set({ value: '2' })}>
              <Typo tag="span" fontSize="xxs" weight="regular">
                12—17h
              </Typo>
            </BorderedButton>
            <BorderedButton
              type="button"
              selected={store.state.value === '3'}
              onClick={() => store.set({ value: '3' })}>
              <Typo tag="div" fontSize="xxs" weight="regular">
                17—22h
              </Typo>
              <Typo tag="strong" fontSize="xxs" weight="bold">
                69,90 €
              </Typo>
            </BorderedButton>
          </ButtonGroup>
          <Typo
            tag="div"
            fontSize="xs"
            weight="regular"
            style={{ marginTop: '12px', marginBottom: '4px' }}>
            Viernes 25.09
          </Typo>
          <ButtonGroup>
            <BorderedButton
              type="button"
              selected={store.state.value === '4'}
              onClick={() => store.set({ value: '4' })}>
              <Typo tag="span" fontSize="xxs" weight="regular">
                12—17h
              </Typo>
            </BorderedButton>
            <BorderedButton
              type="button"
              selected={store.state.value === '5'}
              onClick={() => store.set({ value: '5' })}>
              <Typo tag="div" fontSize="xxs" weight="regular">
                17—22h
              </Typo>
              <Typo tag="strong" fontSize="xxs" weight="bold">
                69,90 €
              </Typo>
            </BorderedButton>
          </ButtonGroup>
        </StyledDivWithTopBorder>
      </div>
    )
  })
)
