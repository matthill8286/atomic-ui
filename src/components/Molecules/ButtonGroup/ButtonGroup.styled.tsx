import { styled, css } from '@/styles'
import { Color, Size } from '@/types'

export type StyledOptionProps = { selected?: boolean }

const color = (x: keyof Color) => ({ theme }) => theme.color[x]
const spacing = (x: keyof Size) => ({ theme }) => theme.spacing.base[x]
const innerShadowColor = (strColor: string) => `inset 0px 0px 0px 1px ${strColor}`
const innerShadowFromTheme = (x: keyof Color) => ({ theme }) => innerShadowColor(theme.color[x])

export const StyledButtonGroup = styled.div`
  display: flex;
  width: 100%;
  background-color: ${color('white')};
  button,
  [type='button'] {
    border-radius: 0;
    flex: 1 1 auto;

    &:first-child {
      border-top-left-radius: 4px;
      border-bottom-left-radius: 4px;
    }
    &:last-child {
      border-top-right-radius: 4px;
      border-bottom-right-radius: 4px;
    }
  }
`

const StyledButton = styled.button.attrs({ type: 'button' })`
  cursor: pointer;
  box-sizing: border-box;
  min-width: 50px;
  padding: ${spacing('xs')};
  border-width: 1px;
  border-style: solid;
  border-color: ${color('grey2')};
  border-radius: ${spacing('xxs')};
  background-color: ${color('white')};

  transition-duration: 0.15s;
  transition-timing-function: ease-in-out;
  transition-property: border-color, box-shadow;

  outline: none;

  &:hover:enabled,
  &:active:enabled {
    box-shadow: ${innerShadowFromTheme('primary')};
    border-color: ${color('primary')};
  }
  &:disabled {
    cursor: not-allowed;
  }
`
export const BorderedButton = styled(StyledButton)<StyledOptionProps>(
  ({ selected, theme }) => css`
    ${Boolean(selected) &&
    css`
      border-color: ${theme.color.primary};
      box-shadow: ${innerShadowColor(theme.color.primary)};
    `}
  `
)
