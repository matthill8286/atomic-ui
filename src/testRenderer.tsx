import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { saiyanTheme } from '@/styles/sc-vars-saiyan'
import { Theme } from '@/types'

export function renderWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>,
  theme = saiyanTheme
) {
  return renderer.create(<ThemeProvider theme={theme}>{tree}</ThemeProvider>).toJSON()
}

export function renderWithThemeAndRouter(tree: React.ReactElement, theme: Theme = saiyanTheme) {
  return renderer
    .create(
      <ThemeProvider theme={theme}>
        <Router>{tree}</Router>
      </ThemeProvider>
    )
    .toJSON()
}

export function renderWithRouter(tree: React.ReactElement) {
  return renderer.create(<Router>{tree}</Router>).toJSON()
}

export function mountWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(<ThemeProvider theme={saiyanTheme}>{tree}</ThemeProvider>)
}

export function mountWithThemeAndRouter<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(
    <ThemeProvider theme={saiyanTheme}>
      <Router>{tree}</Router>
    </ThemeProvider>
  )
}
