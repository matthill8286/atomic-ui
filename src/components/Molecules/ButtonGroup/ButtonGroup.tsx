/* global JSX */
import React from 'react'
import { StyledButtonGroup } from './ButtonGroup.styled'

export function ButtonGroup(props: { children: React.ReactNode }): JSX.Element {
  return <StyledButtonGroup>{props.children}</StyledButtonGroup>
}

ButtonGroup.displayName = 'ButtonGroup'
