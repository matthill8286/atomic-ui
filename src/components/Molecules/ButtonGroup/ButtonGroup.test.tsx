import React from 'react'
import { renderWithTheme } from '@/testRenderer'
import { ButtonGroup } from './ButtonGroup'

describe('ButtonGroup', () => {
  it('renders correctly', () => {
    const component = renderWithTheme(<ButtonGroup>Hello</ButtonGroup>)
    expect(component).toMatchSnapshot()
  })
})
