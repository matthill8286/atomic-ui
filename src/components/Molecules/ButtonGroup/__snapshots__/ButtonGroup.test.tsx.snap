// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  background-color: #fff;
}

.c0 button,
.c0 [type='button'] {
  border-radius: 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c0 button:first-child,
.c0 [type='button']:first-child {
  border-top-left-radius: 4px;
  border-bottom-left-radius: 4px;
}

.c0 button:last-child,
.c0 [type='button']:last-child {
  border-top-right-radius: 4px;
  border-bottom-right-radius: 4px;
}

<div
  className="c0"
>
  Hello
</div>
`;
